recipe/meta.yaml

```
@@ -73,10 +80,14 @@ outputs:
         - mock
         - pytest
         - pip
+        - jax >= 0.3.24  # [osx or linux64]
+        - pytorch-cpu >=1.0  # [not win and not ppc64le]
+        - gym  # [py < 311]
```

```suggestion
        - jax >= 0.3.24  # [osx or linux64]
        # conda-forge doesn't have pytorch on all platforms/arches yet
        - pytorch >=1.0  # [unix and not ppc64le]
        - gym            # [py<311]
```
For context see https://github.com/conda-forge/pytorch-cpu-feedstock/issues/155. Also, where reasonable, I prefer aligned selectors for better readability.

---

In a local test, mamba could not see, that I needed the CPU variant of PyTorch. 🤔 I will check the job output.

---

Ah, sorry, I didn't see that this was only for the test section. Generally, the variant being picked up will depend on whether conda can find the CUDA drivers or not. This will not be the case in CI, but might be the case on your machine.

Background: Hypothetically requiring gymnasium to use the CPU variant would be an extreme restriction to impose on all other packages depending on gymnasium, forcing them to forego the benefits of the CUDA variant for all other purposes even if they have the right hardware - we shouldn't do this under basically any circumstances.

If indeed we want to force the CPU variant for testing, you can add an additional line `pytorch =*=cpu*` to enforce the cpu variant, but I don't think this should be necessary.

---

You are right. The CPU variant was installed in the job. Very good.

I don't remember in what situation pytorch selected the CUDA variant. Maybe it was even from PyPI, when I tested something for gymnasium.